neard/src/runtime.rs

```diff
@@ -294,11 +315,17 @@ impl NightshadeRuntime {
         };
 
         let epoch_height = self.get_epoch_height_from_prev_block(prev_block_hash)?;
+        let epoch_id = {
+            let mut epoch_manager = self.epoch_manager.write().expect(POISONED_LOCK_ERR);
```

Comment:
Can't we use read access here? It is just an extra measure of precaution?

Reply:
`get_block_info` is mutable because of the cache we maintain internally.

Reply:
Got it. It is unfortunate.